fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/FromAdapter.java

```diff
@@ -45,13 +47,71 @@ public FromAdapter from(String tableName, Alias alias) {
         return this;
     }
 
+    public FromAdapter innerJoin(String tableName, Alias alias, WhereFragment joinOnPredicate) {
+        this.select.addInnerJoin(tableName, alias, joinOnPredicate.getExpression());
+        return this;
+    }
+
+    public FromAdapter leftOuterJoin(String tableName, Alias alias, WhereFragment joinOnPredicate) {
+        this.select.addLeftOuterJoin(tableName, alias, joinOnPredicate.getExpression());
+        return this;
+    }
+
+    public FromAdapter innerJoin(Select subQuery, Alias alias) {
+        return this;
+    }
+
     /**
      * Start building the "WHERE" clause for the statement
      * @param predicate
      * @return
      */
     public WhereAdapter where(String predicate) {
-        return new WhereAdapter(this.select, predicate);
+        WhereClause wc = establishWhereClause();
+        return new WhereAdapter(this.select, wc, predicate);
+    }
+
+    /**
+     * Starts building a WHERE clause with the tableAlias/columnName
```

Comment:
```suggestion
     * Starts building a WHERE clause with the {tableAlias}.{columnName}
```

style in other parts

Reply:
ok